PR: https://github.com/odoo/odoo/pull/97988

From: a3eec47f35853a6fa88f94885491f14e7ff94ef6
From: Rob Coekaerts
Date: 2022-09-12 09:38:11

Structural Changes: 3
Total Changes: 19

[IMP] hr_timesheet: Move timesheet_cost into a separate module.

The field timesheet_cost defined on hr_employee in hr_timesheet would be
useful in other modules (eg planning and MRP) as well. Therefore with this
commit the field is moved into a separate module upon which
hr_timesheet will have a dependency. In addition this field is renamed
to employee_hourly_wage, which is a more accurate and neutral name for
this field, as it will not only be used in timesheets.

task-2929518

closes odoo/odoo#97988

Related: odoo/enterprise#30359
Related: odoo/upgrade#3853
Signed-off-by: Kevin Baptiste <kba@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_timesheet/models/__init__.py
+++ b/addons/hr_timesheet/models/__init__.py
@@ -1,7 +1,6 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from . import hr_employee
 from . import hr_timesheet
 from . import ir_http
 from . import ir_ui_menu

--- a/addons/hr_timesheet/models/hr_employee.py
+++ b/None
@@ -1,12 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
-
-
-class HrEmployee(models.Model):
-    _inherit = 'hr.employee'
-
-    timesheet_cost = fields.Monetary('Cost', currency_field='currency_id',
-    	groups="hr.group_hr_user", default=0.0)
-    currency_id = fields.Many2one('res.currency', related='company_id.currency_id', readonly=True)

--- a/addons/hr_timesheet/models/hr_timesheet.py
+++ b/addons/hr_timesheet/models/hr_timesheet.py
@@ -284,7 +284,7 @@ class AccountAnalyticLine(models.Model):
         # (re)compute the amount (depending on unit_amount, employee_id for the cost, and account_id for currency)
         if any(field_name in values for field_name in ['unit_amount', 'employee_id', 'account_id']):
             for timesheet in sudo_self:
-                cost = timesheet._employee_timesheet_cost()
+                cost = timesheet._hourly_cost()
                 amount = -timesheet.unit_amount * cost
                 amount_converted = timesheet.employee_id.currency_id._convert(
                     amount, timesheet.account_id.currency_id or timesheet.currency_id, self.env.company, timesheet.date)
@@ -306,9 +306,9 @@ class AccountAnalyticLine(models.Model):
     def _get_timesheet_time_day(self):
         return self._convert_hours_to_days(self.unit_amount)
 
-    def _employee_timesheet_cost(self):
+    def _hourly_cost(self):
         self.ensure_one()
-        return self.employee_id.timesheet_cost or 0.0
+        return self.employee_id.hourly_cost or 0.0
 
     def _get_report_base_filename(self):
         task_ids = self.task_id
